src/store/api/views.py

```diff
@@ -100,20 +117,46 @@ def post(self, request):
             available=serializer.validated_data["available"],
             cost_price=serializer.validated_data["cost_price"],
         )
-        return Response(
-            ProductSerializer(product).data, status=status.HTTP_201_CREATED
-        )
+        return Response(ProductSerializer(product).data, status=status.HTTP_201_CREATED)
 
 
 class ProductDetailAPIView(generics.RetrieveUpdateDestroyAPIView):
     """
-    View, update, or delete a product.
+    A view for viewing, updating, or deleting a product.
     """
 
     serializer_class = ProductSerializer
     queryset = Product.objects.all()
     permission_classes = (IsAdmin, IsAuthenticated)
 
+    @swagger_auto_schema(
+        operation_description="API endpoint for retrieving a product by ID.",
+        responses={200: openapi.Response("Product details.", ProductSerializer)},
+        operation_id="RetrieveProductByIDStaff",
+    )
+    def get(self, request, *args, **kwargs):
+        return super().get(request, *args, **kwargs)
+
+    @staticmethod
+    def _update_product_swagger_auto_schema(method):
+        """
+        Static method for documenting the put and patch methods.
+        """
+        return swagger_auto_schema(
+            operation_description="API endpoint for updating a product by ID.",
+            request_body=ProductSerializer,
+            responses={200: openapi.Response("Product updated.", ProductSerializer)},
+            operation_id="UpdateProductByIDStaff",
+        )(method)
+
+    @_update_product_swagger_auto_schema
+    def put(self, request, *args, **kwargs):
+        return super().put(request, *args, **kwargs)
+
+    @_update_product_swagger_auto_schema
+    def patch(self, request, *args, **kwargs):
+        return super().patch(request, *args, **kwargs)
+
     def update(self, request, *args, **kwargs):
         instance = self.get_object()
         serializer = self.get_serializer(instance, data=request.data, partial=True)
```

Comment:
why `partial=True`? it should be true only for the `patch` requests, not for put